src/UserRepository/PdoDatabase.php

```diff
@@ -96,15 +111,16 @@ public function authenticate(string $credential, string $password = null) : ?Use
     /**
      * Get the user roles if present.
      *
-     * @param string $identity
-     * @return string[]
+     * @return array<int, mixed>
```

Comment:
This will return a list of strings, thus, I'd prefer keeping `string[]` as it is being inferred as `array<int,string>`. 
This should not become `mixed`!

As the roles supposed to be a list of strings, please change the other typehints regarding roles to list of strings aswell.

Reply:
@boesing, recheck please.